mvp_onboarding.client/src/components/CustomerTable.jsx

```diff
@@ -0,0 +1,354 @@
+import React, { useEffect, useState } from "react";
+import { getCustomers, deleteCustomer, createCustomer, updateCustomer } from "../services/customerService";
+import {
+  Table,
+  Button,
+  Icon,
+  Modal,
+  Form,
+  Input,
+  Pagination,
+  Dropdown
+} from "semantic-ui-react";
+
+const CustomerTable = () => {
+  const [customers, setCustomers] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [deleteOpen, setDeleteOpen] = useState(false);
+  const [newOpen, setNewOpen] = useState(false);
+  const [editOpen, setEditOpen] = useState(false);
+  const [selectedCustomer, setSelectedCustomer] = useState(null);
+  const [createIsDisabled, setCreateIsDisabled] = useState(true);
+  const [editIsDisabled, setEditIsDisabled] = useState(true);
+  const [selectedAddress, setSelectedAddress] = useState("");
+  const [selectedName, setSelectedName] = useState("");
```

Comment:
This is fine. But at some point, you need to decide whether you need to set state for everything or you can set a general one that you can use from a components point of view.